pkg/scheduler/objects/nodesorting.go

```diff
@@ -42,33 +48,79 @@ func (fairnessNodeSortingPolicy) PolicyType() policies.SortingPolicy {
 	return policies.FairnessPolicy
 }
 
-func (binPackingNodeSortingPolicy) ScoreNode(node *Node) float64 {
-	// choose most loaded node first
-	return resources.LargestUsageShare(node.GetAvailableResource())
+func absResourceUsage(node *Node, weights *map[string]float64) float64 {
+	totalWeight := float64(0)
+	usage := float64(0)
+
+	shares := node.GetResourceUsageShares()
+	for k, v := range shares {
+		weight, found := (*weights)[k]
+		if !found || weight == float64(0) {
+			continue
+		}
+		if math.IsNaN(v) {
+			continue
+		}
+		usage += v * weight
+		totalWeight += weight
+	}
+
+	var result float64
+
+	if totalWeight == float64(0) {
+		result = float64(0)
+	} else {
+		result = usage / totalWeight
+	}
+	return result
```

Comment:
can we normalize this value int? I do not think we need that much accuracy using float64. Similar to https://github.com/kubernetes/kubernetes/blob/dd5c3a109e2f156774576b7e7a38e104fa5c7202/pkg/scheduler/framework/interface.go#L100, it has a MaxNodeScore defined.

Reply:
If we are allowing fractional weights (and I think that's imperative) we really do need fractional calculations. Modern CPUs can calculate and operate on 64-bit floats basically as quickly as ints, and we don't do it often enough for this to matter. I've seen optimizations like this in the past result in really weird edge cases where rounding doesn't behave as expected.

Reply:
OK, that also makes sense. Normalization causes extra computes, based on your point: Modern CPUs can calculate and operate on 64-bit floats basically as quickly as ints. I guess we are good for now. Memory is not a major concern right now.